### ISSUE TITLE
Question: redir + http without man-in-the-middle, is it possible?

### ISSUE THREAD

**Issue by julianxhokaxhiu** [nadoo/glider#2]
Hi,

I saw your project and I was astonished and I was trying to replicate the same for myself. Basically what I am trying to archive is something very simple: a full-traffic redirector for HTTP/HTTPS to another proxy in the network.

For the HTTP protocol, actually I found no issue. But for HTTPS I found some headaches. To make it short you can take a look at this article, how he found a partial solution to this topic: http://blog.rchapman.org/posts/Transparently_proxying_any_tcp_connection/

So basically what I would like to know is: as your binary fits perfectly on what I was trying to archive, is it possible to be done through something like:
```bash
# HTTP
glider -l redir://:80 -f http://1.2.3.4:80
# HTTPS
glider -l redir://:443 -f http://1.2.3.4:80
```

Would be definitely awesome, and would also kill the needs to define all the PROXY Environment variables forever!

Thank you in advance,
Julian

**Comment by nadoo** [comment]
Hi Julian, 

The redirect proxy only support on Linux OS now, assume your networking condition like below:
PC -> Gateway(linux box) -> Upstream HTTP Proxy(1.2.3.4:80) -> Internet

First of all, your upstream http proxy MUST support http "CONNECT" method, if so, and you wanna redirect all requests to port 80 and 443 automatically to the upstream proxy, you can:
1. set up a local transparent redirect proxy with glider ON YOUR GATEWAY:
`glider -l redir://:8443 -f http://1.2.3.4:80`

2. redirect all requests to 80 and 443 using iptables:
`iptables -t nat -A PREROUTING -p tcp --dport 80 redir ports 8443`
`iptables -t nat -A PREROUTING -p tcp --dport 443 redir ports 8443`

Then all your PC's requests to port 80 and 443 will auto redirect to glider, and it should work fine.

If you just wanna use glider on your Linux PC, you can run glider on your pc and use the following iptables rules:
`iptables -t nat -A OUTPUT -p tcp --dport 80 -j REDIRECT --to-ports 8443`
`iptables -t nat -A OUTPUT -p tcp --dport 443 -j REDIRECT --to-ports 8443`

Hope I've caught your question, if there's any mistake, please feel free to point out.

**Comment by nadoo** [comment]
I found a bug in http proxy upstream mode and fixed it, so if you will do a test please use the newest version: https://github.com/nadoo/glider/releases

**Comment by julianxhokaxhiu** [comment]
Hi,

first of all *thank you*! I really appreciate the time you spent on my Issue.

Related to the environment: yes, it's Linux machine so i'm quite ok with that. I will try that later/tomorrow maximum and I'll get back to you asap :) The proxy on the other side supports HTTP Connect method, so it should work just fine.

Your project looks really amazing as it combines I guess the best things needed in the proxy world :D Keep pushing forward!

Best regards,
Julian

**Comment by nadoo** [comment]
Thank you! waiting for your good news!  😄
